### PR TITLE
Pr armplatformpkg drop mpcore

### PR DESCRIPTION
**Pull request by ardbiesheuvel**
Drop MPCore variants of PrePi and PrePeiCore, which are obsolete and should no longer be used.

Note that the ArmMpCore PPI is retained - unicore platforms also implement this, and consume it in CpuPei, which copies the info into a HOB that is consumed in other places (ACPI table generation on Ampere, MpServicesDxe). However, this is completely separate from whether or not all cores boot into UEFI or not.

### CODE REVIEW

**Comment by leiflindholm** [comment]
Could we also rename PrePeiCoreUniCore.inf PrePeiCore.inf?

**Comment by ardbiesheuvel** [comment]
> Could we also rename PrePeiCoreUniCore.inf PrePeiCore.inf?

No objections, but this would require a little dance where we clone the INF under its new name, update all users in edk2-platforms and finally drop the old INF.

**Comment by leiflindholm** [comment]
> > Could we also rename PrePeiCoreUniCore.inf PrePeiCore.inf?
> 
> No objections, but this would require a little dance where we clone the INF under its new name, update all users in edk2-platforms and finally drop the old INF.

I'd be happy for it to be done separately from this PR, but would be good to get done within a short-ish span of time.

**Review by leiflindholm — APPROVED** [review]

**Review by samimujawar — APPROVED** [review]
These changes look good to me.

File: ArmPlatformPkg/ArmPlatformPkg.dsc [hunk 1]
```diff
@@ -123,7 +123,6 @@
   ArmPlatformPkg/PrePeiCore/PrePeiCoreMPCore.inf
   ArmPlatformPkg/PrePeiCore/PrePeiCoreUniCore.inf
 
-  ArmPlatformPkg/PrePi/PeiMPCore.inf
```
Comment (leiflindholm) [inline ArmPlatformPkg/ArmPlatformPkg.dsc:126]: Comment actually on commit message (thanks github):
I think the word "from" or suchlike has been dropped before "[mostly]".

File: ArmPlatformPkg/ArmPlatformPkg.dec [hunk 2]
```diff
@@ -50,8 +50,6 @@
   gArmPlatformTokenSpaceGuid   = { 0x9c0aaed4, 0x74c5, 0x4043, { 0xb4, 0x17, 0xa3, 0x22, 0x38, 0x14, 0xce, 0x76 } }
 
 [PcdsFeatureFlag.common]
-  gArmPlatformTokenSpaceGuid.PcdSendSgiToBringUpSecondaryCores|FALSE|BOOLEAN|0x00000004
```
Comment (leiflindholm) [inline ArmPlatformPkg/ArmPlatformPkg.dec:53]: Same commit message comment as previous: looks like some word dropped before "[mostly]"?

File: ArmPlatformPkg/PrePi/PrePi.c [hunk 3]
```diff
@@ -28,6 +28,15 @@
 UINT64  mSystemMemoryEnd = FixedPcdGet64 (PcdSystemMemoryBase) +
                            FixedPcdGet64 (PcdSystemMemorySize) - 1;
 
+/**
```
Comment (leiflindholm) [inline ArmPlatformPkg/PrePi/PrePi.c:31]: Commit message: there's a lot of 'o's in that "tooooools".

**Comment by leiflindholm** [comment]
I had a few minor comments on commit messages and comments, but I've reviewed this set and am very happy about it. This is excellent cleanup!

**Pull request merged**